backend/src/llm_client/openai_client.py

```diff
@@ -0,0 +1,53 @@
+import json
+import openai as OpenAI
+from typing import Optional
+
+from .base import LLMClientBase, MessagePassingInterface
+from .prompt import Prompt
+
+class OpenAIClient(LLMClientBase):
+    def __init__(self, _api_key: str):
+        try:
+            OpenAI.api_key = _api_key
```

Comment:
OpenAI instance is not being initialized correctly, this part of code won't work
It should be initialized as below.
**OpenAI**(_api_key=_api_key_)